pkg/wireguard/bind_test.go

```diff
@@ -0,0 +1,212 @@
+package wireguard_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	corev1alpha "github.com/apoxy-dev/apoxy-cli/api/core/v1alpha"
+	"github.com/apoxy-dev/apoxy-cli/pkg/log"
+	"github.com/apoxy-dev/apoxy-cli/pkg/stunserver"
+	"github.com/apoxy-dev/apoxy-cli/pkg/wireguard"
+	"github.com/apoxy-dev/apoxy-cli/pkg/wireguard/netstack"
+	"github.com/pion/ice/v4"
+	icelogging "github.com/pion/logging"
+	"github.com/pion/stun/v3"
+	"golang.org/x/sync/errgroup"
+	"golang.zx2c4.com/wireguard/conn"
+	"k8s.io/utils/ptr"
+)
+
+func TestICEBind(t *testing.T) {
```

Comment:
Can you also try to do a full e2e test using wireguard device?

Reply:
Will arrive in another branch.